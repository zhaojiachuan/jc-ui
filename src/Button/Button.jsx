import React, { useState } from "react";
import { Button } from "antd";
import _ from "lodash";
const index = ({
    borderColor,
    color,
    bgColor,
    full,
    editType,
    rowMargin,
    columnMargin,
    debounce,
    debounceTime,
    ...props
}) => {
    const [loading, setLoading] = useState(false)
    const testFun = _.debounce(() => {
        console.log(123123)
    }, 1000, {
        'leading': true
    })
    // const onClick = () => {
    //     console.log('123')
    // }
    // 业务按钮样式控制
    let boxStyle = {};
    if (editType === "create") {
        boxStyle.boder = '1px solid #4096ff';
        boxStyle.width = "max-content";
        boxStyle.backgroundColor = `#4096ff`;
        boxStyle.color = '#ffffff';
    } else if (editType === "edit") {
        boxStyle.boder = `1px solid #89d961`;
        boxStyle.width = "max-content";
        boxStyle.backgroundColor = `#89d961`;
        boxStyle.color = '#ffffff';
    } else if (editType === "delete") {
        boxStyle.boder = `1px solid #ff7875`;
        boxStyle.width = "max-content";
        boxStyle.backgroundColor = `#ff7875`;
        boxStyle.color = '#ffffff';
    } else {
        boxStyle.boder = `1px solid ${borderColor}`;
        boxStyle.width = full ? "100%" : "max-content"
        boxStyle.backgroundColor = `${bgColor}`;
        boxStyle.color = `${color}`;
    }
    if (rowMargin) {
        boxStyle.marginLeft = '5px';
        boxStyle.marginRight = '5px';
    }
    if (columnMargin) {
        boxStyle.marginTop = '8px';
        boxStyle.marginBottom = '8px';
    }
    if (debounce) {
        props.onClick = _.debounce(props.onClick, debounceTime ? debounceTime : 1000, {
            'leading': true
        })
    }
    return (
        <Button
            style={boxStyle}
            rowMargin columnMargin {...props}
        >{props.children}</Button>
    );
};
// 没有继承antd样式之前的基本样式
// index.defaultProps = {
//     borderColor: "gray",
//     disabled: false,
//     icon: true,
//     color: "#000000",
//     bgColor: "#ffffff",
//     full: false,
// };

export default index;
