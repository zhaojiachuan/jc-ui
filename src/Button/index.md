---
title: Button 按钮
nav:
  title: 组件
  path: /componments
group:
  title: 基础组件
  path: /basic
  order: 2
---

# Button 按钮

按钮用于开始一个即时操作  
<br/>
## 何时使用


标记了一个（或封装一组）操作命令，响应用户点击行为，触发相应的业务逻辑。
除了支持 ant-design 基本属性，我们还支持用户的部分自定义行为
由于其完全支持antd的基本属性，用户可以前往antd官网查看更多详细文档
<a>https://ant.design/index-cn</a>  
<br/>
为了区分不同业务，除了支持antd原本的五种按钮，JC-UI另外添加了三种不种状态下的按钮
 <ul>
   <li>创建数据按钮</li>
   <li>编辑数据按钮</li>
   <li>删除数据按钮</li>
</ul>
同时支持用户添加相应属性，对按钮进行自定义设计
<br/>
<br/>

## 代码演示

### 基础使用
<code src='./demos/demo1.jsx' ></code>

### 业务按钮

<code src='./demos/demo2.jsx' ></code>
<br/>
通过指定不同的editType，区分不同业务下按钮的不同样式

### 自定义样式
<code src='./demos/demo3.jsx' ></code>
<br/>
对color（字体颜色）、bgColor（背景颜色）等属性赋值，定制个性化按钮

### 自带间距
<code src='./demos/demo4.jsx' ></code>
<br/>
设置rowMargin或columnMargin的值为true，让按钮横向或纵向自带间距

### 防抖按钮
<code src='./demos/demo5.jsx' ></code>
<br/>
具有防抖功能的按钮，设置debounce属性决定是否启用防抖功能，debounceTime参数设置等待时间，默认1s
<br/>
<br/>
<br/>
### 主要参数

| 参数  |                            说明                              |                 类型                  |    默认值   |  版本  |
| :---: | :---------------------------------------------------------: | :-----------------------------------:|   :----:   | :---: |
| editType |              设置editType，切换按钮的三种样式                 |               string                |     -      | 1.0.0 |
| color    |                     字体颜色                               |               string                |   #000000  | 1.0.0 |
| bgColor  |                     背景颜色                               |               string                |   #FFFFFF  | 1.0.0 |
| rowMargin |                    横向间距                               |               boolean               |   false    | 1.0.0 |
| debounce |                     是否开启防抖                            |               boolean               |   false    | 1.0.0 |
| debounceTime |                 函数调用间隔时间                         |               number               |   1000ms    | 1.0.0 |


兼容其他 antd Button 属性
