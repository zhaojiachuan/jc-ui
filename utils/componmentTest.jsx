import React from 'react'
import { render } from '@testing-library/react';

export default function componmentTest (Componment, ComponmentName, ComponmentProps) {
    describe(`component ${ComponmentName}`, () => {
        it(`renders correctly`, () => {
            const wrapper = render(<Componment {...ComponmentProps} />);
            expect(wrapper.container).toMatchSnapshot();
        });
        it('component could be updated and unmounted without errors', () => {
            const { rerender, unmount } = render(<Componment {...ComponmentProps} />);
            expect(() => {
                rerender(<Componment {...ComponmentProps} />);
                unmount();
            }).not.toThrow();
        });
    });
}