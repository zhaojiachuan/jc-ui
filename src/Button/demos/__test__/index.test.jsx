import componmentTest from "../../../../utils/componmentTest";
import React from 'react';
import Button from '../../index';

componmentTest(Button, 'Button', { children: 'button' });