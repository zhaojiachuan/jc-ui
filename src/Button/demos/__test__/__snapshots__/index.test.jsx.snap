// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Button renders correctly 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-mxhywb ant-btn-default"
    type="button"
  >
    <span>
      button
    </span>
  </button>
</div>
`;
